tests/Doctrine/Tests/DBAL/Driver/StatementIteratorTest.php

```diff
@@ -79,9 +82,9 @@ private function assertIterationCallsFetchOncePerStep(Traversable $iterator, int
     }
 
     /**
-     * @return string[][]
+     * @return Generator<class-string<IteratorAggregate&Statement>[]>
      */
-    public static function statementProvider() : iterable
+    public static function statementProvider() : Generator
```

Comment:
Why does it have to be this specific? `Generator` is an implementation of `iterable` of which the caller doesn't care.

`class-string...[]` is misleading. We're not returning a list of class strings. We're returning a tuple, the first argument of which is a class-string. Adding another argument of another type to the tuple will break this expectation.

Reply:
Ok, I'll try another syntax which is less misleading. As for being specific, I usually use as specific return types as I can, it can't hurt, but I will change it back.

Reply:
> I usually use as specific return types as I can

I don't think it's the end goal. The test methods implement the PHPUnit's API that expects data providers to [return a `Traversable`](https://github.com/sebastianbergmann/phpunit/blob/85fefa381541fb3fcbbfe9b5f32be386551cd774/src/Util/Annotation/DocBlock.php#L412). It is fine to be more specific about the contained data type because you want to make sure the items are correct. But there's no point in being more specific about the container type unless you want to create extra coupling between the method signature (`fn (): array`/`fn (): Generator`) and the implementation (`return [];`/`yield ...` respectively).